ts/src/header-validator/trigger.test.ts

```diff
@@ -1097,6 +1098,42 @@ const testCases: jsontest.TestCase<Trigger>[] = [
     ],
   },
 
+  // Attribution Scope
+  {
+    name: 'attribution-scope-not-string',
+    json: `{"attribution_scopes": [1, 2]}`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes', 0],
+        msg: 'must be a string',
+      },
+      {
+        path: ['attribution_scopes', 1],
+        msg: 'must be a string',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scopes-too-many',
+    json: `{"attribution_scopes": ["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21"]}`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes'],
+        msg: 'length must be in the range [0, 20]',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-too-long',
```

Comment:
Please also add test cases for the empty list and something other than a list at all.

Reply:
Done.